roles/openshift_facts/library/openshift_facts.py

```diff
@@ -1187,11 +1189,12 @@ def get_defaults(self, roles):
                         iptables_sync_period='5s', set_node_ip=False)
             defaults['node'] = node
 
-        if 'nfs' in roles:
-            nfs = dict(exports_dir='/var/export', registry_volume='regvol',
-                       export_options='*(rw,sync,all_squash)')
-            defaults['nfs'] = nfs
-
+        defaults['hosted'] = dict(registry_storage_type=None,
+                                  registry_storage_host=None,
+                                  registry_storage_volume_dir='/var/exports',
+                                  registry_storage_volume='registry',
+                                  registry_storage_volume_capacity='5Gi',
+                                  registry_storage_export_options='*(rw,sync,all_squash)')
```

Comment:
Thinking a bit more about this...  we might want more of an object model here...

```
class StorageBase(object):
    def __init__(self, type):
        self.type = type


class PersistentVolume(StorageBase):
    def __init__(self, name, size, create_pv, access_modes)
        StorageBase(self, 'pv')
        self.name = name
        self.size = size
        self.create_pv = create_pv
        self.access_modes = access_modes
        
class NfsPersistentVolume(PersistentVolume):
    def __init__(self, name, size, host, dir, options):
        PersistentVolume(self, name, size, True, [read_write_once, read_write_many, read_only_many])
        self.host = host
        self.dir = dir
        self.options = options
```

Granted classes are a bit heavyweight here given that it would only exist in openshift_facts, but a structured dictionary that extends the idea is probably a good idea to make it easier to add additional types later without just creating a ton of new variables to toss around.